### PR TITLE
Raise informative error when resampling without divisions

### PR DESCRIPTION
**Pull request by mrocklin**
Raised by @plieningerweb on gitter chat

### CODE REVIEW

**Comment by plieningerweb** [comment]
Example: 
```
import pandas
import numpy
from dask.delayed import delayed, compute
import dask.dataframe

@delayed
def get_series(name):
    index = pandas.period_range('2015-01-01', '2015-01-31', freq='d')
    return pandas.DataFrame({'test': numpy.random.randn(20)},index=index[0:20])

dfs = [get_series(f) for f in range(1,3)]
df = dask.dataframe.from_delayed(dfs)
df['test'].resample('1m').mean()
```

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -2958,27 +2959,32 @@ def call(obj, attr, *args):
...
-    mins = index2.map_partitions(lambda x: pd.Series(x.min()), meta=index2)
-    maxes = index2.map_partitions(lambda x: pd.Series(x.max()), meta=index2)
+    result = map_partitions(_set_sorted_index, df, index, drop=drop, meta=meta)
+
+    return compute_divisions(result, **kwargs)
+
+
+def compute_divisions(df, **kwargs):
+    mins = df.index.map_partitions(lambda x: pd.Series(x.min()), meta=df.index)
+    maxes = df.index.map_partitions(lambda x: pd.Series(x.max()), meta=df.index)
```
Comment (nirizr) [inline dask/dataframe/core.py]: Won't `x[0]` and `x[-1]` be better than `x.min()` and `x.max()` here?

It's also a better test for invalid input IMHO

Comment (mrocklin) [inline dask/dataframe/core.py]: The index isn't always sorted within partitions.  I've found it best to play it safe and call min/max.  These are also incredibly cheap, so I don't think that performance wise this is an issue.

**Comment by mrocklin** [comment]
I'd like to merge this soon.  Any further comments?

File: dask/dataframe/io.py [hunk 2]
```diff
@@ -798,9 +798,12 @@ def from_delayed(dfs, meta=None, divisions=None, prefix='from-delayed',
         An iterable of ``dask.delayed.Delayed`` objects, such as come from
         ``dask.delayed`` These comprise the individual partitions of the
         resulting dataframe.
-    $META
```
Comment (jcrist) [inline dask/dataframe/io.py]: Accidental deletion?

Comment (mrocklin) [inline dask/dataframe/io.py]: Why was this here?  I'm not familiar with having `$META` in the docstring.  What is this supposed to mean?

Comment (jcrist) [inline dask/dataframe/io.py]: Instead of copy-pasting around the parameter description of the meta keyword, I made a decorator that inserts it in functions. This was added in #1422.

Comment (mrocklin) [inline dask/dataframe/io.py]: OK, I should have checked `git blame`.  I'll replace it.  

Thoughts on instead using some verbal redirect, like the following:

```python
    meta: pd.DataFrame or pd.Series
        Empty pandas version of the result, see XXX for more information
```

I find myself somewhat irrationally allergic to adding decorators to methods for things like this.  There is some "technology used to benefit gained ratio" test that this fails for me.  It's hard for new developers to reason about what's going on without understanding all of the codebase.

Comment (jcrist) [inline dask/dataframe/io.py]: I also thought about that, but sided with wanting a more verbose description everywhere. Could go either way though. If you want to change it I'd be +1.
